Characters/GameEntities/Humanoid.cs

```diff
@@ -54,4 +59,24 @@ private void UpdateMovementAnimation(CharacterAnimationType animationType)
                 Character.Idle(); break;
         }
     }
+
+    public void ContactCallback(ContactInfo collisionInfo, Dictionary<BodyHandle, ISimulationMember> simulationMembers)
+    {
+        var pair = collisionInfo.CollidablePair;
+        var collidableReference
+            = pair.A.BodyHandle == BodyHandle ? pair.B : pair.A;
+        if (collidableReference.Mobility != CollidableMobility.Dynamic)
+            return;
+#if DEBUG
+        // TODO replace with something more sensible
+        if (simulationMembers.TryGetValue(collidableReference.BodyHandle, out var otherBody))
+        {
+            Console.WriteLine($"Bot collided with {otherBody}");
```

Comment:
Every time a projectile hits a bot I get this message multiple times. Even after adding 
            if (otherBody is Projectile) ((Projectile)otherBody).IsDead = true;
Can we make it so that the hit is only called once? We dont want the bullet to deal damage multiple times

Reply:
This is an issue that I think can't really be solved on the physics engine's side -- we're already listening to events that get fired when a contact is first made so the projectile is either:
* losing contact with a body for a brief period of time
*  "sliding" over the hitbox and that gets interpreted as a new collision 
A solution would be to check the time between two consecutive collisions with the same projectile and if it's unrealistically small ignore the collision